src/main/java/io/github/ImpactDevelopment/installer/target/targets/Vanilla.java

```diff
@@ -52,11 +53,25 @@
     private final String id;
     private final ImpactJsonVersion version;
     private final InstallationConfig config;
+    private final OptiFine optifine;
+    private final Path vanillaJar;
 
-    public Vanilla(InstallationConfig config) {
+    public Vanilla(InstallationConfig config) throws RuntimeException {
         this.version = config.getSettingValue(ImpactVersionSetting.INSTANCE).fetchContents();
+        this.optifine = config.getSettingValue(OptiFineSetting.INSTANCE) ? new OptiFine(config.getSettingValue(OptiFineFileSetting.INSTANCE)) : null;
         this.config = config;
-        this.id = version.mcVersion + "-" + version.name + "_" + version.version + prettifiedOptifineVersion().orElse("");
+        this.id = String.format("%s-%s_%s%s", version.mcVersion, version.name, version.version, optifine == null ? "" : "-OptiFine_"+optifine.getOptiFineVersion());
+
+        // TODO consider downloading the jar from mojang ourselves?
+        //      or at least consider how to do this for multimc?
+        this.vanillaJar = config.getSettingValue(MinecraftDirectorySetting.INSTANCE).resolve("versions").resolve(version.mcVersion).resolve(version.mcVersion+".jar");
+        if (optifine != null && !Files.isRegularFile(vanillaJar)) {
+            throw new IllegalStateException("If installing OptiFine, you must play Minecraft "+version.mcVersion+" at least once before continuing!");
```

Comment:
I think we already have this check later on, don't we? Did you move it?

Reply:
Let's not bait and switch with saying this is needed only for optifine, then require it unconditionally later on lel

Reply:
What do we require it for? Our installer shouldn't touch the vanilla jar?

Reply:
It doesn't touch it, correct. But it does check if it exists. Otherwise ppl were complaining about being unable to launch. 


```
    private void checkVersionInstalled() {
        Path path = config.getSettingValue(MinecraftDirectorySetting.INSTANCE).resolve("versions").resolve(version.mcVersion).resolve(version.mcVersion + ".jar");
        if (!Files.exists(path)) {
            throw new RuntimeException("Please install and run Vanilla " + version.mcVersion + " once as normal before continuing.", new FileNotFoundException(path.toString()));
        }
    }
```

in Vanilla.java

Reply:
Fixed